bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Region.java

```diff
@@ -603,8 +644,39 @@ void translateInPixels (int x, int y) {
 public void translate (Point pt) {
 	if (isDisposed()) SWT.error(SWT.ERROR_GRAPHIC_DISPOSED);
 	if (pt == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
-	pt = DPIUtil.autoScaleUp(pt);
-	translateInPixels(pt.x, pt.y);
+	operations.add(new Operation(OperationType.TRANSLATE, ArgumentType.POINT, new Point(pt.x, pt.y)));
+	Point p = DPIUtil.autoScaleUp(pt, initialZoom);
+	translateInPixels(p.x, p.y);
+}
+
+private HashMap<Integer, Long> handleMap = new HashMap<>();
```

Comment:
Please move that up to the other attributes

Reply:
Okay